src/server/models/db/migrations/20230427161921_add_view_ListsForDatabase/migration.sql

```diff
@@ -0,0 +1,20 @@
+-- This is a migration to help get the correct data for the relevant rows in on the list management dashboard page
+create view "ListsForDashboard" as (
+  select
+    l.id "listId",
+    l.type "type",
+    c.name "country",
+    l."nextAnnualReviewStartDate" "nextAnnualReviewStartDate",
+    l."lastAnnualReviewStartDate" "lastAnnualReviewStartDate",
+    l."jsonData" "jsonData",
+    (case when coalesce(l."nextAnnualReviewStartDate", fp."firstPublished")  + interval '18 months' < current_date then true else false end) as "isOverdue",
+    count(li.*) filter (where li."isPublished") as "live",
+    jsonb_array_length(l."jsonData" -> 'users') as "admins",
+    sum(case when li."status" in ('EDITED', 'CHECK_ANNUAL_REVIEW', 'NEW') then 1 else 0 end) as "actionNeeded"
+  from
+    "List" l
+      inner join "Country" c on  c.id = l."countryId"
+      left outer join "FirstPublishedOnList" fp on fp."listId" = l.id
+      left join "ListItem" li on li."listId" = l.id
```

Comment:
List items won't show up unless `jsonData.metadata.emailVerified` = true. On the test db it appears that there are are a few providers that haven't verified their email.. so our counts will be slightly off. e.g. if it says there are 8 with actionNeeded, but then they view the list items, they may only see 5 listItems in todo. 

I had a check, you can put it in the `sum(..)` but it ends up doubling the query time. CTE keeps it fairly low.


```
create view "ListsForDashboard" as

-- sets up a "temp" table (this is called common table expression CTE). 
with filteredListItem as (select "listId", sum(case when "status" in ('EDITED', 'CHECK_ANNUAL_REVIEW', 'NEW') then 1 else 0 end) actionNeeded, count(*) filter (where "isPublished") live from "ListItem" where "jsonData"->'metadata'->'emailVerified' is not null
                          group by "listId")
-- filteredListItem can now be used as a table for this query

select
    l.id,
    l.type,
    c.name,
    l."nextAnnualReviewStartDate",
    l."lastAnnualReviewStartDate",
    l."jsonData",
    coalesce(l."lastAnnualReviewStartDate", fp."firstPublished") + interval '18 months'  < current_date "isOverdue",
    jsonb_array_length(l."jsonData" -> 'users') as "admins",
    li.live,
    li.actionNeeded
from
    "List" l
        inner join "Country" c on  c.id = l."countryId"
        left outer join "FirstPublishedOnList" fp on fp."listId" = l.id
        join filteredListItem li on li."listId" = l.id
group by l.id, c.name, fp."firstPublished", li.live, li.actionNeeded;
```